azure/services/scalesets/scalesets.go

```diff
@@ -104,7 +105,11 @@ func (s *Service) Reconcile(ctx context.Context) (retErr error) {
 		}
 
 		if fetchedVMSS != nil {
-			s.Scope.SetProviderID(azure.ProviderIDPrefix + fetchedVMSS.ID)
+			providerID, err := azureutil.ConvertResourceGroupNameToLower(azure.ProviderIDPrefix + fetchedVMSS.ID)
```

Comment:
can we add a comment to explain why this is needed (to match the cloud-provider format)?